### PR TITLE
Removed enum instance, added nFromTo

### PR DESCRIPTION
**Pull request by dmcclean**
Proposed implementation of #47.

Somewhat hairier than I expected? Perhaps being overly cautious about numerical issues?

### CODE REVIEW

File: src/Numeric/Units/Dimensional/DK.hs [hunk 1]
```diff
@@ -512,6 +513,18 @@ dimensionlessLength = Dimensional . fromIntegral . length
     length :: Foldable t => t a -> Int
     length = foldl' (\c _ -> c Prelude.+ 1) 0 
 
+-- | Returns a list of quantities between given bounds.
+nFromTo :: (Fractional a, Integral b) => Quantity d a -- ^ The initial value.
+                                      -> Quantity d a -- ^ The final value.
+                                      -> b -- ^ The number of intermediate values. If less than zero, no intermediate values will result.
```
Comment (bjornbm) [inline src/Numeric/Units/Dimensional/DK.hs]: “If less than one” perhaps?

Comment (dmcclean) [inline src/Numeric/Units/Dimensional/DK.hs]: Thanks, I goofed that.

**Comment by bjornbm** [comment]
> Perhaps being overly cautious about numerical issues?

Probably?

Anyway, without changing the numerical properties (as far as I can tell) a first simplification is:

```haskell
nFromTo xi xf n | n <= 0    = [xi, xf]
                | otherwise = let delta = xf - xi
                                  scale s x = realToFrac s *~ one * x
                                  nSteps = n Prelude.+ 1
                               in xi : [xi + scale (i % nSteps) delta | i <- [1..n]] ++ [xf]
```

And to me it is more tractable like this:

```haskell
nFromTo xi xf n | n <= 0    = [xi, xf]
                | otherwise = let delta = xf - xi
                                  nSteps = n Prelude.+ 1
                                  delta_i i = realToFrac (i % nSteps) *~ one * delta
                               in xi : [xi + delta_i i | i <- [1..n]] ++ [xf]
```

I've don't really use list comprehensions a lot and often feel that mapping can be clearer, at least for relatively simple cases. Here:

```haskell
                               in xi : fmap (\i -> xi + delta_i i) [1..n] ++ [xf]
```

Or, which again I don't think will affect the numerics:

```haskell
                               in fmap (\i -> xi + delta_i i) [0..nSteps]
```

I would probably inline `delta`, import `succ` from `Prelude`:

```haskell
nFromTo xi xf n | n <= 0    = [xi, xf]
                | otherwise = let steps = succ n
                                  delta i = realToFrac (i % steps) *~ one * (xf - xi)
                               in fmap (\i -> xi + delta i) [0..steps]
```

Arguably `succ n` could be inlined, and I tend to prefer `where` over `let … in`. I would probably end up with (possibly taking it too far?):

```haskell
nFromTo xi xf n = fmap f [0..succ n]
  where
    f i = xi + realToFrac (i % succ n) *~ one * (xf - xi)
```

Like I said, I don't think any of this changes the numerics. And I wouldn't expect performance to be hurt: It adds a bit of redundant multiplies by zero and one, but removes the conditional and list concatenation. What do you think?

**Comment by bjornbm** [comment]
Merged with my simplification. Let me know if you think it is broken @dmcclean and we can back track!

**Pull request merged**